test/e2e/storage/testsuites/snapshottable.go

```diff
@@ -396,6 +396,93 @@ func (s *snapshottableTestSuite) DefineTests(driver storageframework.TestDriver,
 
 				deleteVolumeSnapshot(ctx, f, dc, sr, pattern, vscontent)
 			})
+
+			ginkgo.It("Verify if online snapshot works", func(ctx context.Context) {
+				init(ctx)
+				// delete the pod at the end of the test
+				ginkgo.DeferCleanup(e2epod.DeletePodWithWait, cs, pod)
+
+				// The pod should be in the Success state.
+				ginkgo.By("[init] check pod success")
+				pod, err = cs.CoreV1().Pods(pod.Namespace).Get(ctx, pod.Name, metav1.GetOptions{})
+				framework.ExpectNoError(err, "Failed to fetch pod: %v", err)
+				framework.ExpectNoError(e2epod.WaitForPodSuccessInNamespaceTimeout(ctx, cs, pod.Name, pod.Namespace, f.Timeouts.PodStartSlow))
+
+				// If test pattern's volumeType is GenericEphemeralVolume or inline volume,
+				// the volumeResource.Pvc is nil, then we need to get the pvc from API server
+				if volumeResource.Pvc == nil {
+					pvcName := ephemeral.VolumeClaimName(pod, &pod.Spec.Volumes[0])
+					pvcNamespace := pod.Namespace
+					pvc, err = cs.CoreV1().PersistentVolumeClaims(pvcNamespace).Get(ctx, pvcName, metav1.GetOptions{})
+					framework.ExpectNoError(err, "get PVC")
+				} else {
+					pvc = volumeResource.Pvc
+				}
+
+				ginkgo.By("creating a snapshot from the source PVC when the pod is using it")
+				parameters := map[string]string{}
+				sr := storageframework.CreateSnapshotResource(ctx, sDriver, config, pattern, pvc.Name, pvc.Namespace, f.Timeouts, parameters)
```

Comment:
See this comment: https://github.com/kubernetes/kubernetes/pull/101280/files#r616971565

Reply:
> See this comment: https://github.com/kubernetes/kubernetes/pull/101280/files#r616971565

> I think the original test is considered more like offline test. The pod was deleted first and then snapshot is taken. The subtle thing here is that even when pod is deleted from api server, there might be a chance that volume is not finished unstaging yet.

So the new online test need to change this logic.

Hi @xing-yang  Do you mean this comment? 
It makes me confused. The snapshot is taken when the pod is running. What do we need to check here?